### PR TITLE
IGAPP-363: Hide NavigationTiles if only Local Information are displayed

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/routes/dashboard/components/__tests__/Dashboard.spec.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  const stateView = new CategoriesRouteStateView(
+    categoryLeaf.path,
+    { [categoryLeaf.path]: categoryLeaf },
+    { [categoryLeaf.path]: [] }
+  )
+  const resourceCache = {}
+  const resourceCacheUrl = 'https://localhost:8080'
+  const t = jest.fn(key => key)
+
+  const createCityModel = (events: boolean, offers: boolean, local: boolean, tunews: boolean, pois: boolean) =>
```
Comment (Taggotty) [inline native/src/routes/dashboard/components/__tests__/Dashboard.spec.js:48]: shouldn't this got to the CityModelsBuilder?

File: native/src/routes/dashboard/components/__tests__/Dashboard.spec.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  })
+
+  it('should not show navigation tiles if there are no features enabled', () => {
+    const cityModel = createCityModel(false, false, false, false, false)
+    const result = TestRenderer.create(renderDashboard(cityModel))
+
+    expect(() => result.root.findByType(NavigationTiles)).toThrowError()
+  })
+
+  it('should show news tile if at least one nems feature is enabled', () => {
```
Comment (Taggotty) [inline native/src/routes/dashboard/components/__tests__/Dashboard.spec.js:102]:
```suggestion
  it('should show news tile if at least one news feature is enabled', () => {
```

File: native/src/modules/categories/components/Categories.js [hunk 3]
```diff
@@ -50,29 +49,25 @@ type PropsType = {|
...
-    navigateToCategory({ cityCode, language, path: tile.path })
+    const { cityModel, language, navigateToCategory } = this.props
+    navigateToCategory({ cityCode: cityModel.code, language, path: tile.path })
   }
 
   onItemPress = (category: { title: string, thumbnail: string, path: string }) => {
-    const { cityCode, language, navigateToCategory } = this.props
-    navigateToCategory({ cityCode, language, path: category.path })
+    const { cityModel, language, navigateToCategory } = this.props
+    navigateToCategory({ cityCode: cityModel.code, language, path: category.path })
```
Comment (Taggotty) [inline native/src/modules/categories/components/Categories.js:58]: Somehow I get an error here, when e.g. navigate to Testumgebung -> Willkommen -> Willkokommen in [Stadt]. It seems that the cityModel is not available here...

**Review by Taggotty — CHANGES_REQUESTED** [review]

File: native/src/routes/dashboard/components/__tests__/Dashboard.spec.js (hunk 1 above)
Comment (klinzo) [reply native/src/routes/dashboard/components/__tests__/Dashboard.spec.js:48]: There is no api in the builders to set custom values yet, so not possible atm.

File: native/src/modules/categories/components/Categories.js (hunk 3 above)
Comment (klinzo) [reply native/src/modules/categories/components/Categories.js:58]: Should be fixed, could you check again please?

**Comment by klinzo** [comment]
This can be tested for example in `Deutschlandvorlage 2019`

File: native/src/routes/dashboard/components/__tests__/Dashboard.spec.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    const result = TestRenderer.create(renderDashboard(cityModel))
+
+    expect(() => result.root.findByType(NavigationTiles)).not.toThrowError()
+    const navigationTiles = result.root.findByType(NavigationTiles)
+    expect(navigationTiles.props.tiles.some(tile => tile.path === 'categories')).toBeTrue()
+  })
+
+  it('should not show navigation tiles if there are no features enabled', () => {
+    const cityModel = createCityModel(false, false, false, false, false)
+    const result = TestRenderer.create(renderDashboard(cityModel))
```
Comment (maxammann) [inline native/src/routes/dashboard/components/__tests__/Dashboard.spec.js:97]: In the future we should try to use the @testing-library as already started in other tests.

File: package.json [hunk 5]
```diff
@@ -20,6 +20,8 @@
   "scripts": {
     "lint": "yarn run eslint",
     "eslint": "eslint .",
+    "test": "yarn --cwd ./native test && yarn --cwd ./web test && yarn --cwd ./api-client test",
+    "flow": "yarn --cwd ./native flow && yarn --cwd ./web flow",
```
Comment (maxammann) [inline package.json:24]:
```suggestion
    "flow": "yarn workspace native flow && yarn workspace web flow",
```

File: package.json [hunk 6]
```diff
@@ -20,6 +20,8 @@
   "scripts": {
     "lint": "yarn run eslint",
     "eslint": "eslint .",
+    "test": "yarn --cwd ./native test && yarn --cwd ./web test && yarn --cwd ./api-client test",
```
Comment (maxammann) [inline package.json:23]:
```suggestion
    "test": "yarn workspace native test && yarn workspace web test && yarn workspace api-client test",
```

File: release-notes/unreleased/IGAPP-363-hide-dashboard.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+issue_key: IGAPP-363
+show_in_stores: false
```
Comment (maxammann) [inline release-notes/unreleased/IGAPP-363-hide-dashboard.yml:2]: I think this should go into the stores as it is a nice feature for cities and users

File: release-notes/unreleased/IGAPP-363-hide-dashboard.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+platforms:
+  - android
+  - ios
+en: Hide redundant local information tile if no other feature is enabled.
```
Comment (maxammann) [inline release-notes/unreleased/IGAPP-363-hide-dashboard.yml:6]: A german entry would be nice here if it is published in stores.

**Review by maxammann — APPROVED** [review]

File: native/src/routes/dashboard/components/__tests__/Dashboard.spec.js (hunk 4 above)
Comment (klinzo) [reply native/src/routes/dashboard/components/__tests__/Dashboard.spec.js:97]: You're definitely right! I somehow thought this is the same library :facepalm:

**Review by maxammann — APPROVED** [review]

File: package.json [hunk 9]
```diff
@@ -20,6 +20,8 @@
   "scripts": {
     "lint": "yarn run eslint",
     "eslint": "eslint .",
+    "test": "yarn workspace native test && yarn workspace web test && yarn workspace api-client test",
```
Comment (Taggotty) [inline package.json:23]: I really like that. :D

**Review by Taggotty — APPROVED** [review]

**Pull request merged**